### ISSUE TITLE
Amazon.Lambda.Tools 5.12.1 breaks application publishing

### ISSUE THREAD

**Issue by martincostello** [aws/aws-extensions-for-dotnet-cli#351]
### Describe the bug

I have a [toy application](https://github.com/martincostello/adventofcode) targeting .NET 9 that is deployed to both AWS Lambda as a custom runtime function and Azure App Service as a container.

A dependabot PR (https://github.com/martincostello/adventofcode/pull/1980) updating Amazon.Lambda.Tools from 5.12.0 to 5.12.1 breaks publishing the application on Linux and Windows with one error and on macOS with a different error.

## Linux and Windows

```text
Amazon Lambda Tools for .NET Core applications (5.12.1)
Project Home: https://github.com/aws/aws-extensions-for-dotnet-cli, https://github.com/aws/aws-lambda-dotnet
	
Host machine architecture (X64) differs from Lambda architecture (Arm64). Building Native AOT Lambda functions require the host and lambda architectures to match.
```

### macOS

```text
Amazon Lambda Tools for .NET Core applications (5.12.1)
Project Home: https://github.com/aws/aws-extensions-for-dotnet-cli, https://github.com/aws/aws-lambda-dotnet
	
No container build image available for targetFramework net9.0 and architecture arm64.
```

The tooling should not block the publish as the application is authored in a way that a working native AoT application is produced that works for an ARM64 custom runtime function.

### Regression Issue

- [X] Select this option if this issue appears to be a regression.

### Expected Behavior

The lambda ZIP archive is produced for deployment.

### Current Behavior

Publishing the application fails.

### Reproduction Steps

1. Clone https://github.com/martincostello/adventofcode/pull/1980/commits/d3e9a00e6ebf4d54836eab9cd59df48128fd1b3f
2. Run `build.ps1 -SkipTests` from the root of the repository

### Targeted .NET platform

.NET 9

### CLI extension version

amazon.lambda.tools 5.12.1

### Environment details (OS name and version, etc.)

Any

**Comment by normj** [comment]
Version 5.12.1 corrected how the tool evaluated MSBuild properties like `PublishAot` to use the `dotnet msbuild -getproperty` command instead of the brute force approach of parsing the csproj file for the XML element `PublishAot`. This was a long time asked feature to be able to handle properties being set in places like `Directory.Build.props`.

In your case you have a `Directory.Build.targets` file that was ignored before 5.12.1 but is not being used in the evaluation of the `PublishAot` property. In that file you are setting `PublishAot` to true which is causing the tool to think it should do a Native AOT compilation.
```
  <PropertyGroup Condition=" ('$(MSBuildProjectName)' == 'AdventOfCode.Console' OR '$(MSBuildProjectName)' == 'AdventOfCode.Site') AND '$(PublishForAWSLambda)' != 'true' ">
    <PublishAot>true</PublishAot>
  </PropertyGroup>
```

This is a change in behavior that I understand the argument is breaking behavior. I would prefer to keep using `dotnet msbuild -getproperty` to evaluate msbuild properties since it is the correct value and people have been asking for this for a long time. Do you think you can change your project around to not attempt `PublishAot`? If you don't want to change your `Directory.Build.targets` file you could add `--msbuild-parameters "/p:PublishAot=false"` in your build.ps1 where you invoke the `dotnet lambda package` command.

**Comment by martincostello** [comment]
I'll try explicitly turning it off, but it should be off by default for publishing the Lambda anyway. The value shouldn't be evaluating as `true` as the condition guarding it should be `false`.

**Comment by martincostello** [comment]
Nope, that didn't work either: https://github.com/martincostello/adventofcode/pull/1980/commits/c8dbd0136d11ba744f2bf6dd09058b6b26a0c39d

The tool is incorrectly determining AoT usage and blocking the publishing.

**Comment by normj** [comment]
@martincostello That is a good catch that we should include the current msbuild parameters specified to `dotnet lambda`. We'll get that fixed. How much of a blocker is this for you? With the US holidays and re:Invent going on next week we are under strict release conditions for non re:Invent related releases. So by default I wouldn't be able to get a fix till after re:Invent.

**Comment by martincostello** [comment]
That's fine - I can easily just ignore the dependabot PR and not take the change. No rush, so post-re:Invent to test any fix is absolutely fine.

**Comment by normj** [comment]
Here is the [PR](https://github.com/aws/aws-extensions-for-dotnet-cli/issues/351) to include the `--msbuild-parameters` value when evaluating MSBuild properties. When we get past our restricted change control for re:Invent I'll get this released.

**Comment by normj** [comment]
Version `5.12.2` has been released with the change. Thanks for helping track the issue down.

**Comment by martincostello** [comment]
Hmm - [it still isn't working](https://github.com/martincostello/adventofcode/pull/2021). I get the same errors as before.

I'll need to look into to see why the change doesn't seem to have made any difference.

**Comment by normj** [comment]
Thanks @martincostello. We'll get the PR released.

**Comment by normj** [comment]
Version

**Comment by normj** [comment]
Version 5.12.3 has been released. Can you let me know if that unblocks you?

**Comment by martincostello** [comment]
All working again now 🚀: https://github.com/martincostello/adventofcode/pull/2029

**Comment by normj** [comment]
Thanks for the confirmation and the help addressing the issue.
